docs/inventory_plugins.md

```diff
@@ -0,0 +1,181 @@
+# Transition to Ansible Inventory Plugins
+Inventory updates change from using scripts which are vendored as executable
+python scripts in the AWX folder `awx/plugins/inventory` (taken originally from
+Ansible folder `contrib/inventory`) to using dynamically-generated
+YAML files which conform to the specifications of the `auto` inventory plugin
+which are then parsed by their respective inventory plugin.
+
```

Comment:
While I understand the desire to have a record of background decisions, I wonder if that is best elsewhere, and we just have a doc that describes what we do and how

Reply:
Also in this folder, we have `scm_inventory.md` and `inventory_refresh.md`. It it tempting to go though and re-organize all of that content.

I agree, _this_ document could be better split into 2 different files. The inventory_refresh doc should also probably be split... maybe into 3 or 4 documents to keep the topics isolated. I would like to create an `inventory` folder and go about re-organizing, but I just didn't want to do that here.